src/dashboard/ui/start.rs

```diff
@@ -0,0 +1,82 @@
+use termion::event::Key;
+
+use crate::common::WrappedRcRefCell;
+use crate::dashboard::events::DashboardEvent;
+use crate::dashboard::state::{DashboardScreen, DashboardState};
+use crate::dashboard::ui::painter::DashboardPainter;
+use crate::dashboard::ui::utils::get_hw_overview;
+use crate::transfer::connection::ClientConnection;
+use std::{io, thread};
+use termion::input::TermRead;
+use tokio::sync::mpsc::UnboundedSender;
+use tokio::time::Duration;
+
+///Starts the dashboard ui with the keyboard listener and tick provider
+pub async fn start_ui_loop(
+    painter: WrappedRcRefCell<DashboardPainter>,
+    state: DashboardState,
+    connection: &mut ClientConnection,
```

Comment:
Let's create the `connection` inside `start_ui_loop`, it doesn't need to be passed to it at this moment.